jetty-util/src/main/java/org/eclipse/jetty/util/thread/PrivilegedThreadFactory.java

```diff
@@ -44,12 +45,24 @@
      */
     static <T extends Thread> T newThread(Supplier<T> newThreadSupplier)
     {
+        // doPriviliged resets the subject
+        // We need to save the subject, and run a subjext.doAs from inside
+        // doPrivileged to restore it.
+        // see https://www.ibm.com/docs/en/was-zos/8.5.5?topic=service-java-authentication-authorization-authorization
+        Subject subject = Subject.getSubject(AccessController.getContext());
```

Comment:
I think this is actually a leak of the `Subject` from the calling thread, no? In your particular case you actually _do_ want to inherit the `Subject` because you have one `Subject` that represents the `server`, but in the general case I don't think we want jetty to inherit anything from from the calling thread.  Let me know if my assumption here is flawed.

I was thinking more along the lines of allowing a ThreadFactory to be set on the `QueuedThreadPool`, rather than always using the `PrivilegedThreadFactory`, in that way you can customize your thread inheritance settings as you like.

Reply:
In my mind the previous behaviour, where the Jetty threads inherited the active subject is the intuitve one, I'd expect the web server threads to run with the same Subject that it was started as.
Others may have other expectations and requirements.

I have tried overriding QueuedThreadPool, but it is not possible because of the private fields, so I ended up copying and renaming the whole class.
Defining a new QueuedThreadPool and using the modified ThreadFactory in it did solve the problems, the tests are running fine.

Being able to set the threadFactory on QueuedThreadPool is a good solution (probably for other purposes too), and it would be a clean and safe way to solve our problem.